exchange/bitswap/wantmanager.go

```diff
@@ -168,11 +168,17 @@ func (mq *msgQueue) runQueue(ctx context.Context) {
 }
 
 func (pm *WantManager) Connected(p peer.ID) {
-	pm.connect <- p
+	select {
+	case pm.connect <- p:
+	case <-pm.ctx.Done():
+	}
 }
```

Comment:
But regardless of why they're useful, I'm +1 on wrapping any potentially blocking activity in a select that also watches the appropriate context.